### PR TITLE
Added NSG

### PR DESCRIPTION
**Pull request by ahems**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Added an NSG locked down to RDP only as per best practices, updated VM type to latest A-series SKU and defaulted to current server OS versions
*
*

### CODE REVIEW

**Comment by ahems** [comment]
I updated the 101-vm-sample-windows template:

- Added an NSG locked down to RDP only
- Updated VM type to latest A-series SKU 
- Defaulted to current server OS version

File: 101-vm-simple-windows/azuredeploy.json [hunk 1]
```diff
@@ -1,187 +1,210 @@
...
-      "defaultValue": "[resourceGroup().location]",
-      "metadata": {
+	"$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
+		"adminUsername": {
+			"type": "string",
+			"minLength": 3,
+			"metadata": {
+				"description": "Username for the Virtual Machine. Value must be at least three characters long and must NOT be any of these values: administrator admin user user1 test user2 test1 user3 admin1 123 actuser adm admin2 aspnet backup console david guest john owner root server sql support support_388945a0 sys test2 test3 user4 user5."
```
Comment (ahems) [inline 101-vm-simple-windows/azuredeploy.json]: Added a better description and used the docs to indicate invalid values

File: 101-vm-simple-windows/azuredeploy.json [hunk 2]
```diff
@@ -1,187 +1,210 @@
...
+			}
+		},
+		"adminPassword": {
+			"type": "securestring",
+			"minLength": 8,
+			"metadata": {
+				"description": "Password for the Virtual Machine - minimum of 8 characters, should have upper and lower case characters, at least one digit, at least one special character. Do NOT use any of these values: abc@123 iloveyou! P@$$w0rd P@ssw0rd P@ssword123 Pa$$word pass@word1 Password! Password1 Password22."
```
Comment (ahems) [inline 101-vm-simple-windows/azuredeploy.json]: Added a more detailed description

File: 101-vm-simple-windows/azuredeploy.json [hunk 3]
```diff
@@ -1,187 +1,210 @@
...
+			"type": "string",
+			"defaultValue": "2019-Datacenter",
+			"allowedValues": [
+				"2008-R2-SP1",
+				"2012-Datacenter",
+				"2012-R2-Datacenter",
+				"2016-Nano-Server",
+				"2016-Datacenter-with-Containers",
+				"2016-Datacenter",
+				"2019-Datacenter"
```
Comment (ahems) [inline 101-vm-simple-windows/azuredeploy.json]: Used latest OS server version as the default

File: 101-vm-simple-windows/azuredeploy.json [hunk 4]
```diff
@@ -1,187 +1,210 @@
...
+			"name": "[variables('vmName')]",
+			"location": "[variables('location')]",
+			"dependsOn": [
+				"[resourceId('Microsoft.Storage/storageAccounts/', variables('storageAccountName'))]",
+				"[resourceId('Microsoft.Network/networkInterfaces/', variables('nicName'))]"
+			],
+			"properties": {
+				"hardwareProfile": {
+					"vmSize": "Standard_A2_v2"
+				},
```
Comment (ahems) [inline 101-vm-simple-windows/azuredeploy.json]: Updated to latest A series version

File: 101-vm-simple-windows/azuredeploy.json [hunk 5]
```diff
@@ -1,187 +1,210 @@
...
+						"destinationPortRange": "3389",
+						"sourceAddressPrefix": "*",
+						"destinationAddressPrefix": "*",
+						"access": "Allow",
+						"priority": 100,
+						"direction": "Inbound"
+					}
+				}]
+			}
+		},
```
Comment (ahems) [inline 101-vm-simple-windows/azuredeploy.json]: Added an NSG with RDP allowed in from the public Internet to better secure this VM

File: 101-vm-simple-windows/azuredeploy.json [hunk 6]
```diff
@@ -1,187 +1,210 @@
...
+						"publicIPAddress": {
+							"id": "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPAddressName'))]"
+						},
+						"subnet": {
+							"id": "[variables('subnetRef')]"
+						}
+					}
+				}],
+				"networkSecurityGroup": {
+					"id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('networkSecurityGroupName'))]"
```
Comment (ahems) [inline 101-vm-simple-windows/azuredeploy.json]: Added NSG to better secure this VM

File: 101-vm-simple-windows/azuredeploy.json [hunk 7]
```diff
@@ -1,187 +1,210 @@
...
+		},
+		{
+			"type": "Microsoft.Network/networkInterfaces",
+			"apiVersion": "2018-11-01",
+			"name": "[variables('nicName')]",
+			"location": "[variables('location')]",
+			"dependsOn": [
+				"[resourceId('Microsoft.Network/publicIPAddresses/', variables('publicIPAddressName'))]",
+				"[resourceId('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]",
+				"[resourceId('Microsoft.Network/networkSecurityGroups', variables('networkSecurityGroupName'))]"
```
Comment (ahems) [inline 101-vm-simple-windows/azuredeploy.json]: Added NSG to better secure this VM

File: 101-vm-simple-windows/azuredeploy.json [hunk 8]
```diff
@@ -1,187 +1,210 @@
...
+	"variables": {
+		"storageAccountName": "[concat(uniquestring(resourceGroup().id), 'sawinvm')]",
+		"nicName": "myVMNic",
+		"addressPrefix": "10.0.0.0/16",
+		"subnetName": "Subnet",
+		"subnetPrefix": "10.0.0.0/24",
+		"publicIPAddressName": "myPublicIP",
+		"vmName": "SimpleWinVM",
+		"virtualNetworkName": "MyVNET",
+		"NetworkSecurityGroupName": "SimpleWinVM-NSG",
```
Comment (ahems) [inline 101-vm-simple-windows/azuredeploy.json]: Added variable with friendly name for new Network Security Group

File: 101-vm-simple-windows/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply 101-vm-simple-windows/azuredeploy.json]: You can add more detail in the desc that says certain values are not allowed, but after that refer to documentation...  the list here will not be complete and create a maint problem.

File: 101-vm-simple-windows/azuredeploy.json (hunk 2 above)
Comment (bmoore-msft) [reply 101-vm-simple-windows/azuredeploy.json]: You can add more detail in the desc that says certain values are not allowed, but after that refer to documentation...  the list here will not be complete and create a maint problem.

File: 101-vm-simple-windows/azuredeploy.json [hunk 9]
```diff
@@ -1,187 +1,210 @@
...
+			"metadata": {
+				"description": "Password for the Virtual Machine - minimum of 8 characters, should have upper and lower case characters, at least one digit, at least one special character. Do NOT use any of these values: abc@123 iloveyou! P@$$w0rd P@ssw0rd P@ssword123 Pa$$word pass@word1 Password! Password1 Password22."
+			}
+		},
+		"dnsLabelPrefix": {
+			"type": "string",
+			"minLength": 3,
+			"maxlength": 24,
+			"metadata": {
+				"description": "Unique DNS Name for the Public IP used to access the Virtual Machine. This value should be at least 3 characters and a maximum of 15."
```
Comment (bmoore-msft) [inline 101-vm-simple-windows/azuredeploy.json]: desc doesn't match the maxLength property

File: 101-vm-simple-windows/azuredeploy.json [hunk 10]
```diff
@@ -1,187 +1,210 @@
...
+		"nicName": "myVMNic",
+		"addressPrefix": "10.0.0.0/16",
+		"subnetName": "Subnet",
+		"subnetPrefix": "10.0.0.0/24",
+		"publicIPAddressName": "myPublicIP",
+		"vmName": "SimpleWinVM",
+		"virtualNetworkName": "MyVNET",
+		"NetworkSecurityGroupName": "SimpleWinVM-NSG",
+		"subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]",
+		"location": "[parameters('location')]"
```
Comment (bmoore-msft) [inline 101-vm-simple-windows/azuredeploy.json]: revert this change and use the parameter directly

File: 101-vm-simple-windows/azuredeploy.json [hunk 11]
```diff
@@ -1,187 +1,210 @@
...
+	},
+	"resources": [{
```
Comment (bmoore-msft) [inline 101-vm-simple-windows/azuredeploy.json]: format the JSON (e.g. SHIFT+ALT+F in VS Code)

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback and conflicts
